edx_toggles/toggles/internal/waffle.py

```diff
@@ -56,13 +56,26 @@ def is_enabled(self, switch_name):
         """
         Returns and caches whether the given waffle switch is enabled.
         """
-        namespaced_switch_name = self._namespaced_name(switch_name)
-        value = self._cached_switches.get(namespaced_switch_name)
+        value = self.get_request_cache(switch_name)
         if value is None:
-            value = switch_is_active(namespaced_switch_name)
-            self._cached_switches[namespaced_switch_name] = value
+            value = switch_is_active(self._namespaced_name(switch_name))
+            self.set_request_cache(switch_name, value)
         return value
 
+    def get_request_cache(self, switch_name, default=None):
```

Comment:
My proposal would be to introduce the following methods:
* get_request_cache(namespaced_switch_name): Use this everywhere it makes sense (if anywhere today).  This is the function we will be moving to.
* get_request_cache_with_short_name(short_switch_name): This is a temporary deprecated function (although you don't have to mark it as such up front).  It just converts the name to a namespaced_name and calls `get_request_cache`.

In this way, get_request_cache will always do the same thing from here on out, and the only change in the future is that more and more code will be using it.  It will never have any backward-incompatible change. 

Backward-incompatible changes are harder to review.  If we went with this PR as-is, I would have to ensure that all calls from `get_request_cache(switch_name)` get updated to `get_request_cache(namespaced_switch_name)` at the same time, at the point in time that the backward-incompatible change is introduced.

The same applies to `set_request_cache` of course.

Reply:
Sounds good, I'll do that.